tests/test_kgain.py

```diff
@@ -0,0 +1,53 @@
+#A file to test the kgain conversion
+import os
+import numpy as np
+from corgidrp.mocks import create_default_headers
+import corgidrp.data as data
+import corgidrp.l2a_to_l2b as l2a_to_l2b
+import pytest
+
+def test_kgain():
+    """
+    test the KGain class and the calibration file and the unit conversion
+    """
+    #test KGain class and cal file
+    prhd, exthd = create_default_headers()
+    dat = np.ones([1024,1024]) * 2 
+    err = np.ones([1,1024,1024]) * 0.5
+    
+    gain_value = np.array([[9.55]])
+    kgain = data.KGain(gain_value, pri_hdr = prhd, ext_hdr = exthd)
+    kgain.save("kgain.fits")
```

Comment:
Are we okay with saving a file to a local user's current directory and then deleting it in line 26?  I've written tests before that create a temp file instead...

Reply:
Maybe @semaphoreP could comment. I could also create a caldb entry for KGain and read it again. The reason that I did this is that I did not want to leave a lot of  files around on the disk.

Reply:
I think this may cause issues for a Windows user.  I think it would be better to use a the `tempfile` module for creating and deleting and cleaning up:
```
import tempfile
tmp_fd, tmp_fits = tempfile.mkstemp(suffix='.fits')
# do stuff with tmp_fits here

os.close(tmp_fd)
os.unlink(tmp_fits)
```

Reply:
Check out some of our other tests (e.g., https://github.com/roman-corgi/corgidrp/blob/main/tests/test_dark_sub.py). We do save files, but we save them inside the `tests` folder, and preferrably inside a subdirectory like `simdata` and `testcalib`. These folders generally won't be tracked, and avoids saving test files in the arbitrary current working directory. 

The use of the `os.path` module should remove any issues in paths between Mac/Windows/Linux.

Reply:
I'm a windows user, and can confirm that the `os` module should be OS agnostic.

Reply:
I remember running into an issue with Windows not with the use of `os` but something about doing a particular thing with a file while it's already opened in Windows vs Mac, and `tempfile` solved the issue for me.  But it looks like it's not applicable in this case then since all this works, and I don't remember the context of my previous issue. :)